File: src/main/java/io/myosotisdev/utopianism/modules/stat/HoldingParticles.java
package io.myosotisdev.utopianism.modules.stat;

import io.myosotisdev.utopianism.modules.stat.data.ItemStatData;
import io.myosotisdev.utopianism.modules.stat.data.ParticleEffectData;
import io.myosotisdev.utopianism.util.Tags;
import net.minestom.server.item.ItemStack;

public class HoldingParticles extends ItemStat
{
    public HoldingParticles()
    {
        super("holding-particles", Tags.createTag("holding-particles", ParticleEffectData.Serializer));
    }

    @Override
    public void onApply(ItemStack itemStack, ItemStatData data)
    {
        return;
    }

    @Override
    public ItemStatData defaultStatData()
    {
        return new ParticleEffectData();
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/stat/HoldingParticles.kt
package io.myosotisdev.utopianism.modules.stat

import com.google.gson.JsonObject
import io.myosotisdev.utopianism.modules.effect.ParticleEffect
import io.myosotisdev.utopianism.modules.stat.data.ItemStatData
import io.myosotisdev.utopianism.modules.stat.data.ParticleEffectData
import io.myosotisdev.utopianism.util.Tags
import net.minestom.server.item.ItemStack
import net.minestom.server.tag.Tag

class HoldingParticles : ItemStat<ParticleEffect>("holding-particles",
        Tags.createTag("holding-particles", ParticleEffectData.Serializer))
{
    override fun onApply(itemStack: ItemStack, data: ItemStatData<out ParticleEffect>)
    {
        return
    }

    override fun defaultStatData(): ItemStatData<ParticleEffect> = ParticleEffectData()

    override fun createStatData(jsonObject: JsonObject): ItemStatData<ParticleEffect> = ParticleEffectData()
}